.github/workflows/ci.yml

```diff
@@ -57,24 +61,25 @@ jobs:
     env:
       # but still use our public caches in any case
       # they might be outdated, in which case a full rebuild will be triggered
-      TARGET_TAG: ${{ github.ref == 'refs/heads/main' && 'alpine' || 'dev-alpine' }}
       CACHE_FROM: |
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-9.5
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-9.6
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-10
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-11
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-12
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-13
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-14
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-15
-        type=registry,ref=pgautoupgrade/pgautoupgrade:build-16
-        type=registry,ref=pgautoupgrade/pgautoupgrade:${{ matrix.pg_target }}-alpine
-        type=registry,ref=pgautoupgrade/pgautoupgrade:${{ matrix.pg_target }}-alpine3.19
-      # we cannot access TARGET_TAG from env
-      # https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-9.5-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-9.6-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-10-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-11-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-12-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-13-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-14-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-15-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:build-16-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:${{ matrix.pg_target }}-${{ matrix.operating_system.flavor }}
+        type=registry,ref=pgautoupgrade/pgautoupgrade:${{ matrix.pg_target }}-${{ matrix.operating_system.flavor }}${{ matrix.operating_system.version }}
```

Comment:
so for `bookworm`, no version is set in the GitHub Actions matrix. If I am not mistaken, this should work as GitHub Actions will substitute the missing `version` as an empty string.

Reply:
Lets try it out. :smile: